src/github.com/matrix-org/dendrite/clientapi/writers/sendmessage.go

```diff
@@ -14,7 +15,9 @@ type SendMessage struct {
 // OnIncomingRequest implements util.JSONRequestHandler
 func (s *SendMessage) OnIncomingRequest(req *http.Request) (interface{}, *util.HTTPError) {
 	logger := req.Context().Value(util.CtxValueLogger).(*log.Entry)
```

Comment:
I realise this isn't part of the PR but would it make sense create a type safe accessors for the logger as per the suggestions in https://golang.org/pkg/context/